### PR TITLE
Factor out `get_company`

### PR DESCRIPTION
**Pull request by alixedi**
### Description of change
This factors out code that searches the company with the given `duns_number` and handles exceptions into a utility function. 

I will be reusing this new function in the upcoming unhappy-path-admin PR.

### Checklist

* [ ] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.md) for instructions
* [ ] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-leeloo/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/dnb_api/utils.py [hunk 1]
```diff
@@ -32,6 +38,47 @@ def search_dnb(query_params):
     return response
 
 
+def get_company(duns_number):
```
Comment (currycoder) [inline datahub/dnb_api/utils.py]: Is it worth making some unit tests for this considering that we are going to use it in a few different locations now?

File: datahub/dnb_api/utils.py [hunk 2]
```diff
@@ -32,6 +38,47 @@ def search_dnb(query_params):
...
+    """
+    Pull data for the company with the given duns_number from DNB and
```
Comment (reupen) [inline datahub/dnb_api/utils.py]: I'm not sure about raising `APIException`s in a utility function that's not just for API views – `APIException`s are just for DRF really.

Comment (alixedi) [reply datahub/dnb_api/utils.py]: I am planning to call this in an admin view. What are you suggesting?

Comment (reupen) [reply datahub/dnb_api/utils.py]: Most probably raising a couple of custom exceptions here (that don't inherit from APIException) e.g. `DNBServiceInternalError`, `DNBServiceNotFoundError`.

Then the API view can translate those to `APIUpstreamException` and `APIBadRequestException`, and the admin view can display the error back to the user.

File: datahub/dnb_api/utils.py [hunk 3]
```diff
@@ -20,6 +19,27 @@
 )
 
 
+class DNBServiceError(Exception):
+    """
+    Exception for when DNB service doesn't return
```
Comment (currycoder) [inline datahub/dnb_api/utils.py]: Minor: Looks like the column width is a bit short here..?

Comment (alixedi) [reply datahub/dnb_api/utils.py]: I get anxious 😂

File: datahub/dnb_api/views.py [hunk 4]
```diff
@@ -163,7 +167,15 @@ def post(self, request):
         duns_serializer.is_valid(raise_exception=True)
         duns_number = duns_serializer.validated_data['duns_number']
 
-        dnb_company = get_company(duns_number)
+        try:
+            dnb_company = get_company(duns_number)
+
+        except (DNBServiceError, DNBServiceInvalidResponse) as exc:
+            raise APIUpstreamException(exc.args[0])
```
Comment (reupen) [inline datahub/dnb_api/views.py]: Would suggest `str(exc)` rather than `exc.args[0]` (same result, but avoids digging into `exc.args` directly).

**Pull request merged**